src/services/expenses.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const { generateIntId } = require('../utils/generateId');
+
+let expenses = [];
+
+function init() {
+  expenses = [];
+}
+
+function getAll(userId, category, from, to) {
+  let foundExpenses = expenses;
+
+  if (userId) {
+    foundExpenses = foundExpenses
+      .filter(expense => expense.userId === +userId);
+  }
+
+  if (category) {
+    foundExpenses = foundExpenses
+      .filter(expense => category.includes(expense.category));
+  }
+
+  if (from) {
+    foundExpenses = foundExpenses
+      .filter(expense => Date.parse(expense.spentAt) >= Date.parse(from));
+  }
+
+  if (to) {
+    foundExpenses = foundExpenses
+      .filter(expense => Date.parse(expense.spentAt) <= Date.parse(to));
+  }
```

Comment:
so if you have all params, you will iterate 4 times?
better to do this logic due to 1 iteration only